modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/baseline/autoadjust/BaselineAutoAdjustScheduler.java

```diff
@@ -55,29 +55,59 @@ public BaselineAutoAdjustScheduler(GridTimeoutProcessor timeoutProcessor,
     }
 
     /**
-     * Adding task to queue with delay and remove previous one.
+     * Adds a new task to queue based on the given {@code baselineAutoAdjustData} with delay and remove previous one.
+     * A new task can be rejected in case of the given {@code baselineAutoAdjustData} is expired or
+     * the target topology version is less than the already scheduled version.
      *
      * @param baselineAutoAdjustData Data for changing baseline.
      * @param delay Delay after which set baseline should be started.
+     * @return {@code true} If a new task was successfully scheduled.
      */
-    public synchronized void schedule(BaselineAutoAdjustData baselineAutoAdjustData, long delay) {
-        if (baselineTimeoutObj != null)
+    public synchronized boolean schedule(BaselineAutoAdjustData baselineAutoAdjustData, long delay) {
+        if (baselineAutoAdjustExecutor.isExecutionExpired(baselineAutoAdjustData)) {
+            if (log.isDebugEnabled())
+                log.debug("Baseline auto adjust data is expired (will not be scheduled) [data=" + baselineAutoAdjustData + ']');
+
+            return false;
+        }
+
+        if (baselineTimeoutObj != null) {
+            long targetVer = baselineAutoAdjustData.getTargetTopologyVersion();
+            long alreadyScheduledVer = baselineTimeoutObj.baselineAutoAdjustData.getTargetTopologyVersion();
+
+            if (alreadyScheduledVer > targetVer) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Baseline auto adjust data is targeted to obsolete version (will not be scheduled) " +
+                        "[data=" + baselineAutoAdjustData + ", scheduled=" + baselineTimeoutObj.baselineAutoAdjustData + ']');
+                }
+
+                return false;
+            }
+
             timeoutProcessor.removeTimeoutObject(baselineTimeoutObj);
+        }
 
-        timeoutProcessor.addTimeoutObject(
+        boolean added = timeoutProcessor.addTimeoutObject(
             baselineTimeoutObj = new BaselineMultiplyUseTimeoutObject(
                 baselineAutoAdjustData,
                 delay, baselineAutoAdjustExecutor,
                 timeoutProcessor,
                 log
             )
         );
+
+        if (log.isDebugEnabled()) {
+            log.info("Scheduled new baseline timeout object [data=" +
```

Comment:
Seems that this log might be wrong if `added` == `false`

Reply:
Thanks! Fixed